tools/integration_tests/util/creds_tests/testdata/revoke_permission_and_creds.sh

```diff
@@ -12,9 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Delete service account after testing
+# Delete key file after testing
 SERVICE_ACCOUNT=$1
 KEY_FILE=$2
+
 gcloud auth revoke $SERVICE_ACCOUNT
-gcloud iam service-accounts delete $SERVICE_ACCOUNT
+# Crete key file output
+# e.g. created key [key_id] of type [json] as [key_file_path] for [service_account]
+# capturing third word from the file to get key-id
+# e.g. capture [KEY_ID]
+KEY_ID=$(cat key_id.txt | cut -d " " -f 3)
+# removing braces
+# e.g. capture KEY_ID
+KEY_ID=${KEY_ID:1:40}
+
+gcloud iam service-accounts keys delete $KEY_ID --iam-account=$SERVICE_ACCOUNT
```

Comment:
isn;t just deleting service account from the project sufficient? why do you need to revoke access and then delete

Reply:
We can create at max 10 keys in service account, so if the limit exceeded it will not create credentials.

Reply:
And the command is deleting only key of the service account not the service account it self.

Reply:
what is the flow here - are we creating a new service account or using the same one in all tests?
Max 10 keys gets applied to deleted keys as well?

Reply:
The flow is.
1. Create service account will create service name creds-test-gcsfuse in the project. If the service account already exist it will use existing one only.
2. create_key_file script will create credentials in that service account for testing.
3. revoke_and_delete_creds will revoke permission and delete the crede which we have created in create_key_file script.

here we are not deleting service account at any place, but accidentally if anyone deleted or we want to run the tests in different project it will create the service account with given name. as we are not deleting service account it can be possible that creation of credentials may exceeded greater than limit(10). which can fail the tests.

Reply:
Please correct me if I am wrong here @Tulsishah 
I think we are doing this so that the service account always exists (not deleted during, before or after the test execution). 
We're creating a key file associated with the service account, using it and then deleting the key file after use (instead of the service account itself) so that it doesn't effect other tests running on other VMs.

Reply:
> what is the flow here - are we creating a new service account or using the same one in all runs? Max 10 keys gets applied to deleted keys as well?

No It will not apply to deleted keys.